### PR TITLE
Add URI count to telemetry Closes #3320

### LINKED ISSUE #3320: Add URI count to telemetry
### Why/User Benefit/User Problem
- product and data scientist need URI count for future A/B testing and validation of regression (similar to Desktop)

/cc @mmccorks

### PR DESCRIPTION
**Pull request by ekager**
Still Needs Data Review

### CODE REVIEW

**Comment by ekager** [comment]
### URI Count

| Event                                       | category | method   | object              | extra           |
|---------------------------------------------|----------|----------|---------------------|-----------------|
| The count of the total non-unique http(s) URIs visited in a subsession, including page reloads, after the session has been restored. This does not include background page requests and URIs from embedded pages or private browsing                    | action   | open    | browser            | `{"total_uri_count": num }`           |
| The count of the unique domains visited in a subsession, after the session has been restored. Subdomains under eTLD are aggregated after the first level (i.e. test.example.com and other.example.com are only counted once). This does not include background page requests and domains from embedded pages or private browsing. The count is limited to 100 unique domains.                      | action   | open    | browser            | `{"unique_domains_count": num }`         |

**Comment by ekager** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?
We will understand if any changes to the product (e.g. GV, other features) will increase/decrease the URI count = corresponding to the engagement with the app.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?

Determine whether changes to the product increase engagement and quality of the app for users.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
We have access to the number of sessions, but we need to understand engagement on a more micro level.

4) Can current instrumentation answer these questions?
No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>"total_uri_count"
The count of the total non-unique http(s) URIs visited in a subsession, including
page reloads, after the session has been restored. This does not include background
page requests and URIs from embedded pages or private browsing</td>
    <td>Category 2 / Interaction data</td>
    <td></td>
  </tr>
 <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>unique_domains_count:
The count of the unique domains visited in a subsession, after the session
has been restored. Subdomains under eTLD are aggregated after the first level
(i.e. test.example.com and other.example.com are only counted once).
This does not include background page requests and domains from embedded pages
or private browsing.</td>
    <td>Category 2 / Interaction data</td>
    <td></td>
  </tr>
</table>


6) How long will this data be collected?  Choose one of the following:

* I want to permanently monitor this data. @bbinto 

7) What populations will you measure?

* Which release channels?
Beta/Nightly/Release

* Which countries?
All 

* Which locales?
All

* Any other filters?  Please describe in detail below.
Only users with telemetry enabled.

8) If this data collection is default on, what is the opt-out mechanism for users?
Telemetry is on by default in Focus and off by default in Klar. Users can opt-out in Settings.

9) Please provide a general description of how you will analyze this data.
Look at number of URIs visited over time to see if it increases with improvements to the app.

10) Where do you intend to share the results of your analysis?
Within Mozilla

**Comment by ekager** [comment]
Assigning @liuche for data review

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 1]
```diff
@@ -294,9 +300,13 @@ object TelemetryWrapper {
     }
 
     private var histogram = IntArray(HISTOGRAM_SIZE)
+    private var domainMap = HashSet<String>()
+    private var numUri = 0
 
     @JvmStatic
-    fun addLoadToHistogram(newLoadTime: Long) {
+    fun addLoadToHistogram(url: String, newLoadTime: Long) {
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: Looks like we're storing this url in order to count distinct values, but we only upload the count.

@ekager can you add a comment that the keys of this map should never be uploaded to telemetry?

**Pull request merged**